### PR TITLE
[styling] PEP8, Mypy and styling changes

### PR DESCRIPTION
**Pull request by eyalf-st**
Mostly PEP8 and Mypy (typing) related touch-ups as well as some PyCharm-Inspection touch-ups and finally a few additional ones.

Also, due to full type-hinting support, had to expose/import "ndarray" from `numpy` which is used in OpenCV's APIs. `numpy` was anyway an indirect dependency and not it has become a direct dependency. Pipefile(s) have been updated and `requirements.txt` has been re-generated (`via pipenv lock -r > requirements.txt`).

In high-level, these are the main (PEP8/Mypy) changes:
  1) Handing import in general-to-specific order
  2) Removed exception handling for imports of built-ins and internal modules
  3) Made static-able methods `@staticmehtod`(s)
  4) Add Type-Hinting
  5) Shorten too-long lines
  6) Many "x,y,z" cases replaced with "x, y, z"

**NOTE1:** After this PR, code now is PyCharm-Inspection and Mypy warning/error free
**NOTE2:** PR is based on https://github.com/kkoomen/qbr/pull/6 due to the need to update the `numpy` dependency in the Pipfile(s) and re-generate the `requirements.txt`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,19 +30,55 @@ That inspired me to create my own. I started using images only and eventually sw
...
+#### Installing pipenv: ([Full Guide](https://pypi.org/project/pipenv/))
+```
+$ sudo apt install pipenv
+```
+
+NOTE: If you have (and need also) an earlier than 3.7 version of python installed, you might need to used [pyenv](https://github.com/pyenv/pyenv-installer) to manage multiple python versions on your machine.
+
 # Installation
-Start off by cloning:
+###Start off by cloning:
```
Comment (kkoomen) [inline README.md:43]: Add a whitespace in between `###` and the word `Start` and add a new empty line before and after every heading. Please do this for all headings.

**Review by kkoomen — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (eyalf-st) [reply README.md]: Fixed (in the base review - https://github.com/kkoomen/qbr/pull/6)!

**Comment by eyalf-st** [comment]
BTW, my suggestion is to do the review for https://github.com/kkoomen/qbr/pull/6 and once it is in, i can merge it into this review/branch and you will get a clean PR with only styling related changes (e.g. README was changed in the other review, where pipenv was introduced).

**Review by kkoomen — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -28,19 +30,67 @@ That inspired me to create my own. I started using images only and eventually sw
...
+
+You will need to install `pipenv` on your machine as well as have python 3.7 installed.
+
+#### Installing pipenv: ([Full Guide](https://pypi.org/project/pipenv/))
+
+```
+$ sudo apt install pipenv
+```
+
+NOTE: If you have (and need also) an earlier than 3.7 version of python installed, you might need to used [pyenv](https://github.com/pyenv/pyenv-installer) to manage multiple python versions on your machine.
```
Comment (kkoomen) [inline README.md:43]: `You might need to used [pyenv] to manage ...` -> `You might need to use [pyenv] in order to manage ..`

File: src/colordetection.py [hunk 3]
```diff
@@ -5,22 +5,29 @@
...
 
+# Built-ins
+from sys import exit as die
+from typing import Tuple
 
-from sys import exit as Die
+# 3rd Party
 try:
-    import sys
+    from numpy import ndarray as np_ndarray     # type: ignore
```
Comment (kkoomen) [inline src/colordetection.py:14]: Can `# type: ignore` be removed? If not, please remove leading spaces in between `np_ndarray` and `# type: ignore`

File: src/colordetection.py [hunk 4]
```diff
@@ -36,33 +43,32 @@ def get_color_name(self, hsv):
...
-            'blue'   : (255,0,0),
-            'green'  : (0,255,0),
-            'white'  : (255,255,255),
-            'yellow' : (0,255,255)
+            'red':      (0, 0, 255),
+            'orange':   (0, 165, 255),
+            'blue':     (255, 0, 0),
+            'green':    (0, 255, 0),
+            'white':    (255, 255, 255),
+            'yellow':   (0, 255, 255)
```
Comment (kkoomen) [inline src/colordetection.py:60]: Remove leading spaces:

```
color = {
    'red':    (0, 0, 255),
    'orange': (0, 165, 255),
    'blue':   (255, 0, 0),
    'green':  (0, 255, 0),
    'white':  (255, 255, 255),
    'yellow': (0, 255, 255)
}
```

File: src/qbr.py [hunk 5]
```diff
@@ -5,62 +5,70 @@
...
+# 3rd Party
 try:
-    import sys
-    import kociemba
-    import argparse
-
-    from combiner import combine
-    from video import webcam
-    from normalizer import normalize
+    import kociemba     # type: ignore
```
Comment (kkoomen) [inline src/qbr.py:15]: Can `# type: ignore` be removed? If not, please remove leading spaces in between `np_ndarray` and `# type: ignore`

File: src/video.py [hunk 6]
```diff
@@ -5,25 +5,33 @@
...
+from sys import exit as die
+from typing import List, Dict, Optional
 
-from sys import exit as Die
+# 3rd Party
 try:
-    import sys
-    import cv2
-    from colordetection import ColorDetector
+    import cv2                                  # type: ignore
```
Comment (kkoomen) [inline src/video.py:14]: Can `# type: ignore` be removed? If not, please remove leading spaces in between `np_ndarray` and `# type: ignore`

File: src/video.py [hunk 7]
```diff
@@ -76,16 +86,16 @@ def color_to_notation(self, color):
         :param color: the requested color
         """
         notation = {
-            'green'  : 'F',
-            'white'  : 'U',
-            'blue'   : 'B',
-            'red'    : 'R',
-            'orange' : 'L',
-            'yellow' : 'D'
+            'green':    'F',
```
Comment (kkoomen) [inline src/video.py:89]: Remove leading spaces:

```
notation = {
    'green':  'F',
    'white':  'U',
    'blue':   'B',
    'red':    'R',
    'orange': 'L',
    'yellow': 'D'
}
```

File: src/video.py [hunk 8]
```diff
@@ -97,13 +107,13 @@ def scan(self):
...
-                   'white','white','white',
-                   'white','white','white']
-        state   = [0,0,0,
-                   0,0,0,
-                   0,0,0]
+        sides = {}
+        preview = ['white', 'white', 'white',
+                   'white', 'white', 'white',
+                   'white', 'white', 'white']
+        state = ['', '', '',
```
Comment (kkoomen) [inline src/video.py:114]: Why empty strings?

**Review by kkoomen — CHANGES_REQUESTED** [review]

**Comment by kkoomen** [comment]
Closing due to no response and I've updated master anyway with a refactored version.

**Pull request closed without merging**